marconi-chain-index/src/Marconi/ChainIndex/Indexers/ExtLedgerStateCoordinator.hs

```diff
@@ -0,0 +1,757 @@
+
+extLedgerConfig :: Lens.Getter (ExtLedgerStateCoordinator input) ExtLedgerConfig
+extLedgerConfig = genesisConfig . Lens.to CE.mkExtLedgerConfig
+
+snapshotInterval :: Lens.Getter (ExtLedgerStateCoordinator input) Word
+snapshotInterval = extLedgerStateCoordinatorConfig . configSnapshotInterval
+
+securityParam :: Lens.Getter (ExtLedgerStateCoordinator input) SecurityParam
+securityParam = extLedgerStateCoordinatorConfig . configSecurityParam
+
+extractBlockEvent :: Lens.Getter (ExtLedgerStateCoordinator input) (input -> BlockEvent)
+extractBlockEvent = extLedgerStateCoordinatorConfig . configExtractBlockEvent
+
+blocksToNextSnapshot :: Lens' (ExtLedgerStateCoordinator input) Word
+blocksToNextSnapshot = extLedgerStateCoordinatorState . stateBlocksToNextSnapshot
+
+snapshots :: Lens' (ExtLedgerStateCoordinator input) [C.ChainPoint]
+snapshots = extLedgerStateCoordinatorState . stateSnapshots
+
+lastEpochNo :: Lens' (ExtLedgerStateCoordinator input) (Maybe C.EpochNo)
+lastEpochNo = extLedgerStateCoordinatorState . stateEpochNo
+
+mkExtLedgerStateCoordinator
+  :: (MonadIO m, MonadError Core.IndexerError m, Core.Point (ExtLedgerStateEvent, input) ~ C.ChainPoint)
+  => ExtLedgerStateCoordinatorConfig (ExtLedgerStateEvent, input)
+  -> FilePath
+  -> [Core.Worker (ExtLedgerStateEvent, input) C.ChainPoint]
+  -> m (ExtLedgerStateCoordinator (ExtLedgerStateEvent, input))
+mkExtLedgerStateCoordinator config rootDir workers = do
+  let genesisCfg = config ^. configGenesisConfig
+      securityParam' = config ^. configSecurityParam
+      extLedgerCfg = CE.mkExtLedgerConfig genesisCfg
+      configCodec = O.configCodec . O.getExtLedgerCfg $ extLedgerCfg
+  liftIO $ createDirectoryIfMissing True rootDir
+  epochStateIndexer' <-
+    buildExtLedgerStateEventIndexer
+      configCodec
+      securityParam'
+      (rootDir </> "epochState")
+  epochBlocksIndexer <-
+    buildBlockIndexer
+      configCodec
+      securityParam'
+      (rootDir </> "epochBlocks")
+  coordinator' <- liftIO $ Core.mkCoordinator workers
+  let extLedgerStateCoordinatorState' =
+        ExtLedgerStateCoordinatorState
+          (config ^. configSnapshotInterval)
+          Nothing
+          []
+  pure $
+    ExtLedgerStateCoordinator
+      config
+      extLedgerStateCoordinatorState'
+      epochStateIndexer'
+      epochBlocksIndexer
+      coordinator'
+
+data ExtLedgerStateWorkerConfig m input = ExtLedgerStateWorkerConfig
+  { workerEventExtractor :: input -> BlockEvent
+  , workerLogger :: BM.Trace m Text
+  , workerNodeConfigPath :: FilePath
+  , workerSnapshotInterval :: Word
+  , workerSecurityParam :: SecurityParam
+  }
+
+{-
+  p <- max <$> Core.lastStablePoint epochStateIndexer' <*> Core.lastStablePoint epochBlocksIndexer
+  ledgerState <-
+    either
+      (const $ throwError $ Core.IndexerInternalError "Can't resolve the last stable point")
+      pure
+      =<< runExceptT (restoreLedgerStateFromIndexers genesisCfg p epochStateIndexer' epochBlocksIndexer)
+-}
```

Comment:
Should this be removed?

Reply:
Indeed, legacy of the last changes I made to the module.